pkg/kn/traffic/compute_test.go

```diff
@@ -51,169 +53,188 @@ func newTestTrafficCommand() (*cobra.Command, *flags.Traffic) {
+			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("", "", 100, true), newTarget("", "rev-v1", 0, false)),
+			[]string{"--traffic", "@latest=10%,rev-v1=90%"},
+			[]string{"@latest", "rev-v1"},
+			[]string{"", ""},
 			[]int{10, 90},
 		},
-		// add 2 new revisions with tags without giving them traffic portions
 		{
+			"add 2 more tagged revisions without giving them traffic portions",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "", 100, true)),
 			[]string{"--tag", "@latest=current,echo-v0=stale,echo-v1=old"},
 			[]string{"@latest", "echo-v0", "echo-v1"},
 			[]string{"current", "stale", "old"},
 			[]int{100, 0, 0},
 		},
-		// re-assign same tag to echo-v1
 		{
+			"re-assign same tag to 'echo-v1' revision",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 100, false)),
 			[]string{"--tag", "echo-v1=latest"},
 			[]string{"echo-v1"},
 			[]string{"latest"},
 			[]int{100},
 		},
-		// set 2% to @latest (append it in traffic block)
 		{
+			"set 2% traffic to latest revision by appending it in traffic block",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 100, false)),
 			[]string{"--traffic", "@latest=2,echo-v1=98"},
 			[]string{"echo-v1", "@latest"},
 			[]string{"latest", ""},
 			[]int{98, 2},
 		},
-		// set 2% to @latest with tag (append it in traffic block)
 		{
+			"set 2% to @latest with tag (append it in traffic block)",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 100, false)),
 			[]string{"--traffic", "@latest=2,echo-v1=98", "--tag", "@latest=testing"},
 			[]string{"echo-v1", "@latest"},
 			[]string{"latest", "testing"},
 			[]int{98, 2},
 		},
-		// change traffic percent of an existing revision in traffic block, add new revision with traffic share
 		{
+			"change traffic percent of an existing revision in traffic block, add new revision with traffic share",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("v1", "echo-v1", 100, false)),
 			[]string{"--tag", "echo-v2=v2", "--traffic", "v1=10,v2=90"},
 			[]string{"echo-v1", "echo-v2"},
 			[]string{"v1", "v2"},
 			[]int{10, 90}, //default value,
 		},
-		// append new non-@latest target
-		{
-			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("v1", "echo-v1", 100, false)),
-			[]string{"--tag", "echo-v2=v2", "--traffic", "v1=10,v2=90"},
-			[]string{"echo-v1", "echo-v2"},
-			[]string{"v1", "v2"},
-			[]int{10, 90}, //default value,
-		},
-		// untag
 		{
+			"untag 'latest' tag from 'echo-v1' revision",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 100, false)),
 			[]string{"--untag", "latest"},
 			[]string{"echo-v1"},
 			[]string{""},
 			[]int{100},
 		},
+		{
+			"untag 'latest' tag from '@latst' revision",
+			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "", 100, true)),
```

Comment:
this test doesn't check anything additional TBH as untagging is unconditional whether its latest or non-latest tag.

I tried to keep the tests compact to avoid more tests, the coverage is already 100%

as we don't have latest revision check in Untag method, I'd suggest we more this test as its just duplicate of earlier one.

Reply:
> as we don't have latest revision check in Untag method,

Ah, that makes sense.